pycontrails/core/datalib.py

```diff
@@ -505,6 +526,19 @@ def open_metdataset(
         :func:`xarray.open_mfdataset`
         """
 
+    @abc.abstractmethod
+    def set_met_source_metadata(self, ds: xr.Dataset | MetDataset) -> None:
+        """Set met source metadata on ``ds.attrs``.
+
+        This is called within the :meth:`open_metdataset` method to set metadata
+        on the returned :class:`MetDataset` instance.
+
+        Parameters
+        ----------
+        met : xr.Dataset | MetDataset
```

Comment:
`ds` is the parameter here, not `met`

Reply:
c9c2957fc4659033c477a6745c5c7bc11f12a531